src/main/java/com/segment/analytics/android/integrations/amplitude/AmplitudeIntegration.java

```diff
@@ -136,6 +169,60 @@ public void identify(IdentifyPayload identify) {
     }
   }
 
+  private void incrementAndSetTraits(Traits traits) {
+    Identify identify = new Identify();
+
+    for (Map.Entry<String, Object> entry : traits.entrySet()) {
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      if (traitsToIncrement.contains(key)) {
+        if (value instanceof Double) {
+          double doubleValue = (Double) value;
+          identify.add(key, doubleValue);
+        }
+        if (value instanceof Float) {
+          float floatValue = (Float) value;
+          identify.add(key, floatValue);
+        }
+        if (value instanceof Integer) {
+          int intValue = (Integer) value;
+          identify.add(key, intValue);
+        }
+        if (value instanceof Long) {
+          long longValue = (Long) value;
+          identify.add(key, longValue);
+        }
+        if (value instanceof String) {
+          String stringValue = String.valueOf(value);
+          identify.add(key, stringValue);
+        }
+      } else {
+        if (value instanceof Double) {
```

Comment:
Similar to earlier, it might be better to split this block into it's own function, say `setTrait(key, value)`

Reply:
Cool - will merge for now and come back to refactor after I get a better idea of how setOnce will fit into this.